tests/unit/test_bugjira.py

```diff
@@ -0,0 +1,180 @@
+from unittest.mock import Mock, create_autospec
+from xmlrpc.client import Fault
+
+import pytest
+from bugzilla import Bugzilla
+from jira import JIRA
+from jira.exceptions import JIRAError
+
+import bugjira.broker as broker
+from bugjira.bugjira import Bugjira
+from bugjira.issue import BugzillaIssue, JiraIssue
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup(monkeypatch):
+    """Patch out the bugzilla.Bugzilla and jira.JIRA constructors
+    in the broker module so that we don't attempt to connect to an actual
+    backend
+    """
+    monkeypatch.setattr(broker, "Bugzilla", create_autospec(Bugzilla))
+    monkeypatch.setattr(broker, "JIRA", create_autospec(JIRA))
+
+
+@pytest.fixture(scope="function")
+def sandboxed_bugjira(good_config_dict):
+    return Bugjira(config_dict=good_config_dict)
+
+
+def test_init_with_config_dict(good_config_dict):
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with a valid config dict
+    THEN the resulting Bugjira instance's config attribute will be the dict
+        we input to the constructor
+    AND the broker and backend attributes should be present
+    """
+    bugjira = Bugjira(config_dict=good_config_dict)
+    assert bugjira.config == good_config_dict
+    assert bugjira._bugzilla_broker
+    assert bugjira._jira_broker
+    assert bugjira.bugzilla
+    assert bugjira.jira
+
+
+def test_init_with_config_path(good_config_file_path, good_config_dict):
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with a path to a known good config file
+    THEN the resulting instance's config attribute should be equivalent to the
+        contents of the config file
+    AND the broker and backend attributes should be present
+    """
+    bugjira = Bugjira(config_path=good_config_file_path)
+    assert bugjira.config == good_config_dict
+    assert bugjira._bugzilla_broker
+    assert bugjira._jira_broker
+    assert bugjira.bugzilla
+    assert bugjira.jira
+
+
+def test_init_with_no_parameters():
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with no parameters
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira()
+
+
+def test_init_with_only_bz_backend_supplied():
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with only the bugzilla parameter supplied
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira(bugzilla=Mock())
+
+
+def test_init_with_only_jira_backend_supplied():
+    """
+    GIVEN the Bugjira class' constructor
+    WHEN we call it with only the jira parameter supplied
+    THEN a BrokerInitException should be raised
+    """
+    with pytest.raises(broker.BrokerInitException):
+        Bugjira(jira=Mock())
+
```

Comment:
Since you have shown an Exception on only bugzilla and only jira, should there also be a test to show valid for when both are passed in with no config?